src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,248 @@
+        if (deadlineDescription.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        String[] deadlineArgs = deadlineDescription.split(" /by ");
+        Task deadline = new Deadline(deadlineArgs[0], deadlineArgs[1]);
+        taskList.add(deadline);
+        updateStorage();
+        return String.format(ADD_TASK_MESSAGE, deadline, taskList.size());
+    }
+
+    /**
+     * Adds a Event task to TaskList.
+     * @param eventDescription The description of the Event task.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the description is blank.
+     */
+    public String addEventTask(String eventDescription) {
+        if (eventDescription.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        String[] eventArgs = eventDescription.split(" /from ");
+        String[] eventTime = eventArgs[1].split(" /to ");
+        String startTime = eventTime[0];
+        String endTime = eventTime[1];
+        Task event = new Event(eventArgs[0], startTime, endTime);
+        taskList.add(event);
+        updateStorage();
+        return String.format(ADD_TASK_MESSAGE, event, taskList.size());
+    }
+
+    /**
+     * Marks a task in TaskList as completed.
+     * @param taskIndex The index of the task in TaskList.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the taskIndex is missing.
+     */
+    public String markTask(String taskIndex) {
+        if (taskIndex.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        Task t = taskList.get(Integer.parseInt(taskIndex) - 1);
+        t.markAsDone();
+        updateStorage();
+        return String.format(MARK_DONE_MESSAGE, t);
+    }
+
+    /**
+     * Unmarks a task in TaskList as not completed.
+     * @param taskIndex The index of the task in TaskList.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the taskIndex is missing.
+     */
+    public String unmarkTask(String taskIndex) {
+        if (taskIndex.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        Task t = taskList.get(Integer.parseInt(taskIndex) - 1);
+        t.markAsUndone();
+        updateStorage();
+        return String.format(MARK_UNDONE_MESSAGE, t);
+    }
+
+    /**
+     * Deletes a task in TaskList.
+     * @param taskIndex The index of the task to be deleted in TaskList.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the taskIndex is missing.
+     */
+    public String deleteTask(String taskIndex) {
+        if (taskIndex.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        Task taskToRemove = taskList.get(Integer.parseInt(taskIndex) - 1);
```

Comment:
Should `Integer.parseInt(taskIndex)` be wrapped in a try-catch? Since it is possible for `taskIndex` to not be in the expected format, throwing a `NumberFormatException`?